components/script/dom/windowproxy.rs

```diff
@@ -381,6 +383,99 @@ impl WindowProxy {
+        let is_feature_sep = |c: char| ['\t', '\n', '\x0c', '\r', ' ', '=', ','].contains(&c);
+        // Step 1
+        let mut tokenized_features = HashMap::new();
+        // Step 2
+        let mut iter = features.chars();
+        let mut cur = iter.next();
+
+        // Step 3
+        while cur != None {
+            // Step 3.1 & 3.2
+            let mut name = String::new();
+            let mut value = String::new();
+            // Step 3.3
+            while let Some(cur_char) = cur {
+                if !is_feature_sep(cur_char) {
+                    break;
+                }
+                cur = iter.next();
+            }
+            // Step 3.4
+            while let Some(cur_char) = cur {
+                if is_feature_sep(cur_char) {
+                    break;
+                }
+                name.push(cur_char.to_ascii_lowercase());
+                cur = iter.next();
+            }
+            // Step 3.5
+            let normalized_name = String::from(match name.as_ref() {
+                "screenx" => "left",
+                "screeny" => "top",
+                "innerwidth" => "width",
+                "innerheight" => "height",
+                _ => name.as_ref(),
+            });
+            // Step 3.6
+            while let Some(cur_char) = cur {
+                if cur_char == '=' || cur_char == ',' || !is_feature_sep(cur_char) {
+                    break;
+                }
+                cur = iter.next();
+            }
+            match cur {
+                // Step 3.7
+                Some(_cur_char) if is_feature_sep(_cur_char) => {
+                    // Step 3.7.1
+                    while let Some(cur_char) = cur {
+                        if !is_feature_sep(cur_char) || cur_char == ',' {
+                            break;
+                        }
+                        cur = iter.next();
+                    }
+                    // Step 3.7.2
+                    while let Some(cur_char) = cur {
+                        if is_feature_sep(cur_char) {
+                            break;
+                        }
+                        value.push(cur_char.to_ascii_lowercase());
+                        cur = iter.next();
+                    }
+                },
+                _ => {},
+            }
+            // Step 3.8
+            if !name.is_empty() {
+                tokenized_features.insert(normalized_name, value);
+            }
+        }
+        // Step 4
+        tokenized_features
+    }
+
+    // https://html.spec.whatwg.org/multipage/#concept-window-open-features-parse-boolean
+    fn parse_open_feature_boolean(
+        tokenized_features: &HashMap<String, String>,
+        name: &str,
+    ) -> bool {
+        match tokenized_features.get(name) {
+            None => false,
+            Some(value) => match value.as_ref() {
+                "" => true,
+                "yes" => true,
+                _ => match value.parse::<i32>() {
```

Comment:
With regards to the failing test you mention, instead of using `parse`, you could implement a parsing function as described at https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#rules-for-parsing-integers

I suspect that would help with getting the test to pass.

Reply:
Thanks for the advice. I guess I'll need to implement my own parser for this.

Another concern I have here is about the value range, which is not mentioned in the above standard. One potential solution is to just implement a procedure to tell whether a string represents zero, non-zero or invalid number, which is just what we need here and can guarantee correctness. Or is there other place in servo that I can take as reference?

Reply:
Perhaps if you were to try to use `value.parse::<f32>()` you would get a better result? If the problem is that `1.5` returns false, using https://doc.rust-lang.org/std/primitive.f32.html might help.

This might be a better first step as opposed to implementing the parser right now...

Reply:
Well I think it's not the problem of interger or float. OTOH, I believe the standard is saying that if a string _begins with_ a valid integer, it's okay. Not only "1.5" is true, even "1a5" should be interpreted as 1 and thus true.

I tested with firefox and chrome just now. And found that they apparently haven't implement tokenization :-(

Reply:
You're right, looking at https://infra.spec.whatwg.org/#collect-a-sequence-of-code-points (used within https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#rules-for-parsing-integers), I read that the parsing should basically stop a `1`, ignore `.5`(or anything else after a non-digit character), and return true. 

So I guess this will require an actual integers parser implementation.

Would you be interested in adding that to your current implementation?

Reply:
@jdm Are you aware of any such parsing code somewhere in the codebase?

Reply:
@gterzian Yep! There's `parse_integer` in `components/style/attr.rs`.